pkg/Microsoft.Private.PackageBaseline/packageIndex.json

```diff
@@ -1258,7 +1258,7 @@
       "BaselineVersion": "4.0.0",
       "AssemblyVersionInPackageVersion": {
         "4.0.0.0": "4.0.0",
-        "4.0.1.0": "4.3.0"
+        "4.1.0.0": "4.3.0"
```

Comment:
cc: @ericstj 

What exactly does the 'packageIndex.json' file do?  What are the rules for when we touch this file?  Do we modify this for everything contract revision we plan to do?